sycl/test/basic_tests/accessor/accessor_assignment_operator.cpp

```diff
@@ -0,0 +1,62 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+
+// Testing each of the specialised assignment operators of 0 dimensional
+// buffer accessors, local accessors, and host accessors.
+
+#include <sycl/sycl.hpp>
+
+int main() {
+  sycl::queue Q;
+
+  {
+    int Data = 0;
+
+    using AccT = sycl::accessor<int, 0, sycl::access::mode::read_write>;
+    sycl::buffer<int, 1> Buf(&Data, sycl::range<1>(1));
+
+    Q.submit([&](sycl::handler &CGH) {
+      sycl::accessor<int, 0> A(Buf, CGH);
+      CGH.single_task<class KernelAccessor>([A] {
+        typename AccT::value_type DataToWrite = 1;
+        A = DataToWrite;
+      });
+
+      assert(A == 1);
+    });
+    Q.wait();
+  }
+
+  {
+    using AccT = sycl::local_accessor<int, 0>;
+    Q.submit([&](sycl::handler &CGH) {
+      AccT LA{CGH};
+      CGH.single_task<class KernelLocalAccessor>([LA] {
+        typename AccT::value_type DataToWrite = 2;
+        LA = DataToWrite;
+      });
+
+      assert(LA == 2);
+    });
+    Q.wait();
+  }
+
+  {
+    int Data = 0;
+
+    using AccT = sycl::host_accessor<int, 0, sycl::access::mode::read_write>;
+    sycl::buffer<int> Buf(&Data, sycl::range<1>(1));
+
+    Q.submit([&](sycl::handler &CGH) {
+      AccT HA(Buf);
```

Comment:
It doesn't seem to be legal to create a host accessor here and access it in the kernel.

Reply:
Resolved this with new unit test: [sycl/unittests/accessor/HostAccessorAssignmentOperator.cpp](https://github.com/intel/llvm/pull/8069/files#diff-d39ed331362bfea08d0e7d92fc0ffa7f155e9ceeb790d85f901ce48e3532da46)